File: examples/05-production/go/worker_options_manual.go
package production

import (
	"context"

	"go.temporal.io/sdk/client"
	"go.temporal.io/sdk/worker"
	"go.temporal.io/sdk/workflow"
)

// Manual Worker sizing. Go has no thread-per-activity model (it uses goroutines),
// so the concurrency knobs are slot counts on worker.Options — the direct
// analogue of Java's setMaxConcurrentActivityExecutionSize / WorkflowTask sizes.
func StartManuallySizedWorker(c client.Client) worker.Worker {
	w := worker.New(c, "io-heavy", worker.Options{
		MaxConcurrentActivityExecutionSize:     200,
		MaxConcurrentWorkflowTaskExecutionSize: 20,
	})
	w.RegisterWorkflow(IoWorkflow)
	w.RegisterActivity(IoActivity)
	return w
}

func IoWorkflow(ctx workflow.Context) error { return nil }

func IoActivity(ctx context.Context) error { return nil }

File: examples/05-production/python/worker_options_manual.py
from temporalio.client import Client
from temporalio.worker import Worker

# Manual Worker sizing. The Python SDK runs activities on an asyncio event loop
# (and a thread/process pool for sync activities) rather than JVM threads, so the
# knobs are concurrency limits rather than thread-pool sizes. These are the direct
# analogue of Java's setMaxConcurrentActivityExecutionSize / WorkflowTask sizes.


async def start(client: Client) -> None:
    worker = Worker(
        client,
        task_queue="io-heavy",
        workflows=[],  # register IoWorkflow here
        activities=[],  # register IoActivities here
        max_concurrent_activities=200,
        max_concurrent_workflow_tasks=20,
    )
    await worker.run()
